EduVPN-redesign/Controllers/iOS/ConnectionInfoViewController.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  ConnectionInfoViewController.swift
+//  EduVPN
+//
+
+import UIKit
+
+class ConnectionInfoViewController: UITableViewController, ParametrizedViewController {
+    enum ConnectionInfoRow: Int {
+        case duration = 0
+        case dataTransferred = 1
+        case address = 2
+        case profileName = 3
+
+        var title: String {
+            switch self {
+            case .duration: return NSLocalizedString("DURATION", comment: "")
+            case .dataTransferred: return NSLocalizedString("DATA TRANSFERRED", comment: "")
+            case .address: return NSLocalizedString("ADDRESS", comment: "")
+            case .profileName: return NSLocalizedString("PROFILE", comment: "")
+            }
+        }
+
+        func value(from connectionInfo: ConnectionInfoHelper.ConnectionInfo) -> String {
+            switch self {
+            case .duration: return connectionInfo.duration
+            case .dataTransferred: return connectionInfo.dataTransferred
+            case .address: return connectionInfo.addresses
+            case .profileName: return connectionInfo.profileName ?? ""
+            }
+        }
+    }
+
+    struct Parameters {
+        let connectionInfo: ConnectionInfoHelper.ConnectionInfo
+    }
+
+    var connectionInfo: ConnectionInfoHelper.ConnectionInfo? {
+        didSet(oldValue) {
+            // The profile name doesn't change, so we don't reload that.
+            // If the address has changed, reload it, else leave it as it is,
+            // so that the copy context menu doesn't get dismissed.
+            if oldValue?.addresses == connectionInfo?.addresses {
+                tableView.reloadRows(indices: [0, 1])
```

Comment:
Perhaps use ConnectionInfoRow.duration.rawValue etc. instead so its clearer what happens here?

Reply:
Fixed in commit 5b962bb.